pkg/proxy/winkernel/proxier.go

```diff
@@ -348,14 +362,30 @@ func (proxier *Proxier) endpointsMapChange(oldEndpointsMap, newEndpointsMap prox
 	// This will optimize remote endpoint and loadbalancer deletion based on the annotation
 	var svcPortMap = make(map[proxy.ServicePortName]bool)
 	var logLevel klog.Level = 5
+	proxier.remoteEPsToCleanup = make(map[string]bool)
+
 	for svcPortName, eps := range oldEndpointsMap {
 		logFormattedEndpoints("endpointsMapChange oldEndpointsMap", logLevel, svcPortName, eps)
+		if proxier.winProxyEbpfMode {
+			for _, ep := range eps {
+				if !ep.IsLocal() && !proxier.remoteEPsToCleanup[ep.IP()] {
+					proxier.remoteEPsToCleanup[ep.IP()] = true
+				}
+			}
+		}
 		svcPortMap[svcPortName] = true
 		proxier.onEndpointsMapChange(&svcPortName, false)
 	}
 
 	for svcPortName, eps := range newEndpointsMap {
 		logFormattedEndpoints("endpointsMapChange newEndpointsMap", logLevel, svcPortName, eps)
+		if proxier.winProxyEbpfMode {
+			for _, ep := range eps {
+				if !ep.IsLocal() && proxier.remoteEPsToCleanup[ep.IP()] {
```

Comment:
Probably isLocal check is not needed here. ep in remoteEPsToCleanup shouldn't be local per check in line 371. And if at all ep is local and in remoteEPsToCleanup, we would ideally like to remove it from remoteEPsToCleanup?

Reply:
These changes are no more part of pr.